graph/collection-properties.md

```diff
@@ -0,0 +1,402 @@
+OData services treat collections of complex types differently than they treat collections of entity types. 
+This is due to the nature of entity types being "individually addressable" (they have some key which uniquely identifies them within the collection) while complex types are not individually addressable. 
+The result of this difference is that it is almost always preferable to use entity types for collections rather than complex types; please see the exceptions [here](#exceptions).
+Let's use the following CSDL as an example:
+
+```xml
+<EntityType Name="interestingData">
+  <Property Name="foos" Type="Collection(self.foo)" Nullable="false" />
+  <NavitationProperty Name="bars" Type="Collection(self.bar)" Nullable="false" ContainsTarget="true" />
+</EntityType>
+
+<ComplexType Name="foo">
+  <Property Name="someProperty" Type="Edm.String" Nullable="false" />
+</ComplexType>
+
+<EntityType Name="bar">
+  <Key>
+    <PropertyRef Name="id" />
+  </Key>
+  <Property Name="id" Type="Edm.String" Nullable="false" />
+  <Property Name="differentProperty" Type="Edm.Int32" Nullable="false" />
+</EntityType>
+```
+
+## Retrieving the elements in a collection
+
+The [OData](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#_Toc31358935) [standard](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#_Toc31358947) specifies `foos` and `bars` both can be retrieved using a `GET` request:
+> OData services support requests for data via HTTP GET requests.
+> 
+> ...
+> 
+> OData services support querying collections of entities, complex type instances, and primitive values.
+
+### Complex Type
+
+```HTTP
+GET  /interestingData/foos
+
+200 OK
+{
+  "value": [
+    {
+      "someProperty": "an original value"
+    }
+  ]
+}
+```
+
+### Entity Type
+
+```HTTP
+GET  /interestingData/bars
+
+200 OK
+{
+  "value": [
+    {
+      "id": "firstBarId",
+      "differentProperty": 10
+    },
+    {
+      "id": "secondBarId",
+      "differentProperty": -6914
+    }
+  ]
+}
+```
+
+## Adding individual elements to a collection
+
+For both `foos` and `bars`, the OData standard specifies that elements can be added to the collection using a `POST` request
+1. [Complex Types](https://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_UpdateaCollectionProperty):
```

Comment:
The protocol supports this, but it's a bit asymmetric to support adding but not removing, so I'm not sure it's a pattern we want to encourage on ms graph. at least we should all out that it's an uncommon pattern due to this asymmetry.